textile/features.textile

```diff
@@ -321,6 +321,7 @@ h3(#rest-channel). RestChannel
 ** @(RSL6a)@ All messages received will be decoded automatically based on the @encoding@ field and the payloads will be converted into the format they were originally sent using i.e. binary, string, or JSON
 *** @(RSL6a1)@ A set of tests must exist to ensure that the client library provides data encoding & decoding interoperability with other client libraries. The tests must use the "set of predefined interoperability message fixtures":https://github.com/ably/ably-common/blob/main/test-resources/messages-encoding.json to 1) publish a raw message to the REST API using the JSON transport and subscribe to the message using Realtime to ensure the @data@ attribute matches the fixture; 2) publish a message using the REST client library and retrieve the raw message using the history REST API using the JSON transport ensuring the @data@ matches the fixture; 3) perform the client library operation using both @JSON@ and @MsgPack@ transports. For reference, see the "Ruby":https://github.com/ably/ably-ruby/pull/94 and "iOS":https://github.com/ably/ably-cocoa/pull/459 implementations
 *** @(RSL6a2)@ A set of tests must exist to ensure that the client library provides interoperability for the @extras@ field which is a JSON-encodable object (ie a value that represents a JSON @object@ value and supports serialization to and from JSON text). The test, at a minimum, should publish a message with an @extras@ object such as @{"push":[{"title":"Testing"}]}@ and ensure it is received with an equivalent JSON-encodable object
+*** @(RSL6a3)@ A set of tests should exist to ensure that the client library can successfully encode and decode binary encoded protocol-messages. The tests should use the "set of predefined interoperability msgpack protocol-message fixtures":https://github.com/ably/ably-common/blob/main/test-resources/msgpack_test_fixtures.json. The msgpack field in this fixture represents the base64 encoding of the entire protocol-message, but the "data" field represents the first element of the messages field of the protocol-message.
```

Comment:
This spec mentions tests for encoding binary messages but, as far as I can tell, the shared fixtures are specifically meant for testing decoding - if we were to use the same fixtures for encoding we would presumably need the fixtures to include all `Message`s of the decoded protocol message, not just the first.
```suggestion
*** @(RSL6a3)@ A set of tests should exist to ensure that the client library can successfully decode binary encoded protocol-messages. The tests should use the "set of predefined interoperability msgpack protocol-message fixtures":https://github.com/ably/ably-common/blob/main/test-resources/msgpack_test_fixtures.json. The msgpack field in this fixture represents the base64 encoding of the entire protocol-message, but the "data" field represents the first element of the messages field of the protocol-message.
```

I also think it would be nice to explain how the fixtures should be used in more detail. For example, how should the `numRepeat` and `encoding` fields be used in the tests, if at all?

Reply:
Also some minor formatting suggestions:
```suggestion
*** @(RSL6a3)@ A set of tests should exist to ensure that the client library can successfully encode and decode binary encoded protocol messages. The tests should use the "set of predefined interoperability msgpack protocol-message fixtures":https://github.com/ably/ably-common/blob/main/test-resources/msgpack_test_fixtures.json. The @msgpack@ field in this fixture represents the base64 encoding of the entire @ProtocolMessage@, but the @data@ field represents the first element of the @ProtocolMessage#messages@ array.
```